js/libs/popover.js

```diff
@@ -115,7 +115,21 @@ var $ = require('jquery'),
   }
 
   $(document).ready(function() {
-    $('[data-toggle="popover"]').popover();
+    $('[data-toggle="popover"], [rel="clickover"]')
+    .on('click', function(e) {
+      e.preventDefault();
+      e.stopPropagation();
+    }).popover();
+  });
+
+  $(document).on('click', function (e) {
+    $('[data-toggle="popover"][data-autoclose="true"], [rel="clickover"]').each(function () {
+        //the 'is' for buttons that trigger popups
+        //the 'has' for icons within a button that triggers a popup
+        if (!$(this).is(e.target) && $(this).has(e.target).length === 0 && $('.popover').has(e.target).length === 0) {                
```

Comment:
I'm fairly certain this will cover all of these conditions, `closest` checks against the element it's being invoked on and then walks up the tree, so in this instance `e.target` will not be `this` or `.popover` and it will also not be a descendant of those nodes.

```if ($(e.target).closest($(this, '.popover')).length === 0)```